find.lua

```diff
@@ -64,93 +64,99 @@ local bwdDataAlgoNames = {
 
 local algoNames = {fwdAlgoNames, bwdFilterAlgoNames, bwdDataAlgoNames}
 
-local function call(layer, f, ...)
-   if find.verbose then
+local function getConvolutionDescriptor(desc)
+   local CUDNN_DIM_MAX=4
+
+   local data = {
+      dim_p = ffi.new('int[1]'),
+      padA = ffi.new('int[?]', CUDNN_DIM_MAX),
+      filterStrideA = ffi.new('int[?]', CUDNN_DIM_MAX),
+      upscaleA = ffi.new('int[?]', CUDNN_DIM_MAX),
+      mode_p = ffi.new('cudnnConvolutionMode_t[1]'),
+      math_p = ffi.new('cudnnDataType_t[1]')
+   }
+
+   cudnn.errcheck('cudnnGetConvolutionNdDescriptor', desc[0], CUDNN_DIM_MAX,
+                  data.dim_p, data.padA, data.filterStrideA,
+                  data.upscaleA, data.mode_p, data.math_p)
+
+   data.arrayLength = data.dim_p[0]
+   data.mode =     data.mode_p[0]
+   data.dataType = data.math_p[0]
+   return data
+end
 
-        print("find:call: calling " .. f .. ", hash: ",  layer.autotunerHash)
+local function verboseCall(layer, f, ...)
+   if find.verbose then
+        print("find:verboseCall: calling " .. f .. ", hash: ",  layer.autotunerHash)
    end
    local status = cudnn.call(f, ...)
    if status ~= ffi.C.CUDNN_STATUS_SUCCESS and (find.verbose or find.verboseError) then
-      local stride = ffi.new('int[8]')
-      local upscale = ffi.new('int[8]')
-      local dim = ffi.new('int[8]')
-      local mode = ffi.new('cudnnConvolutionMode_t[8]')
-      local datatype = ffi.new('cudnnDataType_t[8]')
-      cudnn.call('cudnnGetConvolutionNdDescriptor', layer.convDesc[0],
-                 4, dim, pad, stride,
-                 upscale, mode, datatype)
-      print("find:call:" .. f .. " failed: ", tonumber(status) , ' mode : ', tonumber(mode[0]), ' datatype : ', tonumber(datatype[0]))
+      local desc= cudnn.getConvolutionDescriptor(layer.convDesc)
+      print("find:verboseCall:" .. f .. " failed: ", tonumber(status) , ' mode : ',
+            desc.mode, ' datatype : ', desc.datatype)
    end
    if find.verbose then
-      print("find:call: success, " .. f )
+      print("find:verboseCall: success, " .. f )
    end
    return status
 end
-find.call = call
+find.verboseCall = verboseCall
```

Comment:
Why do you need find.verboseCall (as opposed to local verboseCall)? Is it used anywhere ouside of find.lua?